### PR TITLE
Add XOr operation to socketMask

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR adds XOr support for socketMask

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @derekwaynecarr

File: pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go [hunk 1]
```diff
@@ -137,6 +137,36 @@ func TestOr(t *testing.T) {
 	}
 }
 
+func TestXOr(t *testing.T) {
+	tcases := []struct {
+		name          string
+		firstMaskBit  int
+		secondMaskBit int
+		orMask        string
```
Comment (mattjmcnaughton) [inline pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go]:
```suggestion
		xorMask        string
```

Can we call this `xorMask` instead? I think its clearer.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

One small request around variable naming. Feel free to ping me after its fixed to re-add the /lgtm label.

**Comment by tedyu** [comment]
@mattjmcnaughton 
You comment has been addressed.

**Comment by mattjmcnaughton** [comment]
/lgtm

adding after change

**Comment by tedyu** [comment]
ping @derekwaynecarr for review

**Comment by tedyu** [comment]
@derekwaynecarr 
Mind taking a look ?

thanks

**Comment by tedyu** [comment]
@vishh 
Can you take a look ?

**Comment by tedyu** [comment]
@yujuhong 
Please take a look.

**Comment by tedyu** [comment]
@ConnorDoyle
Can you take a look ?

**Comment by klueska** [comment]
I'm OK with adding this change in general (for completeness), but I'm also curious of the immediate benefit.

The `SocketMask` abstraction (in its current form) is an internal abstraction only meant for use by the `TopologyManager`. Since the TopologyManager doesn't require `XOr` from a `SocketMask` yet, I'm curious what inspired this change so quickly.

**Comment by tedyu** [comment]
This is mostly for completeness.

We can park the PR until scenario involving Xor comes up.

**Pull request closed without merging**